x-pack/plugins/security_solution/common/endpoint/schema/actions.ts

```diff
@@ -70,14 +71,29 @@ export const ActionDetailsRequestSchema = {
 export const EndpointActionListRequestSchema = {
   query: schema.object({
     agentIds: schema.maybe(
-      schema.arrayOf(schema.string({ minLength: 1 }), { minSize: 1, maxSize: 50 })
+      schema.oneOf([
+        schema.arrayOf(schema.string({ minLength: 1 }), { minSize: 1, maxSize: 50 }),
+        schema.string({ minLength: 1 }),
```

Comment:
Why are we introducing this? I'm sure I've missed something, but it's weird having an `agentIds` param accepting a single string, what about having an array with only one agent id?
Same for the `commands` and `userIds` ones below.

Reply:
Also, having the same key in the url multiple times is something we could avoid I guess:
`http://localhost:5601/api/endpoint/action?agentIds=abc&agentIds=def`

Reply:
Yeah, so we had issues with the earlier schema version. I'm almost quoting @paul-tavares here 😄 

"The query params parser does not actually know that `agentIds` is supposed to be an array when only 1 value was sent. So it parses it to a string like `{ agentIds: 'some-id' }`. So when that is given as input to the schema, it craps out (rightfully so) because `agentIds` is defined as an array. ".
 
In order for the parser to not crap out, we had to stringify all `string[]` inputs, and that was not pretty.

We altered the schema so we don't have to stringify the query input and the parser would correctly process the input.